PowerPlan: Powercfg.exe instead of WMI and allow GUID as power plan name

J0F3 opened the pull request:
<!--
    Thanks for submitting a Pull Request (PR) to this project.
    Your contribution to this project is greatly appreciated!

    Please prefix the PR title with the resource name,
    e.g. 'ResourceName: My short description'.
    If this is a breaking change, then also prefix the PR title
    with 'BREAKING CHANGE:',
    e.g. 'BREAKING CHANGE: ResourceName: My short description'.

    You may remove this comment block, and the other comment blocks, but please
    keep the headers and the task list.
-->
#### Pull Request (PR) description
Changes the PowerPlan resource so that is uses powercfg.exe to get and set power plans instead of the WMI method which is not available or working on all Windows version. This would fix the problems on Windows 10, Windows Server 2019 and Nano Server.
Furthermore the resource accepts now also a GUID as the power plan "name" so the desired plan can be als specified as a GUID when using it on a non-english Windows. 

#### This Pull Request (PR) fixes the following issues
<!--
    If this PR does not fix an open issue, replace this comment block with None.
    If this PR resolves one or more open issues, replace this comment block with
    a list the issues using a GitHub closing keyword, e.g.:
-->
- Fixes #59 
- Fixes #155 
- Fixes #65 


#### Task list
<!--
    To aid community reviewers in reviewing and merging your PR, please take
    the time to run through the below checklist and make sure your PR has
    everything updated as required.

    Change to [x] for each task in the task list that applies to your PR.
    For those task that don't apply to you PR, leave those as is.
-->
- [x] Added an entry under the Unreleased section of the change log in the CHANGELOG.md.
      Entry should say what was changed, and how that affects users (if applicable).
- [ N/A] Resource documentation added/updated in README.md in the resource folder.
- [x] Resource parameter descriptions added/updated in schema.mof
      and comment-based help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [N/A ] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] Integration tests added/updated (where possible). See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [x] New/changed code adheres to [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/computermanagementdsc/202)
<!-- Reviewable:end -->

J0F3 commented on appveyor.yml:
```diff
@@ -9,7 +9,7 @@ install:
         $mainModuleFolder = "Modules\$moduleName"
         Import-Module "$env:APPVEYOR_BUILD_FOLDER\DscResource.Tests\AppVeyor.psm1"
-        Invoke-AppveyorInstallTask -PesterMaximumVersion 4.0.8
+        Invoke-AppveyorInstallTask
```
Why was pester restricted to version 4.0.8? Had do remove that because my tests were failing because of the too old Pester version. Is that a problem?

PlagueHO replied:
All good - we were planning on removing this. There was a bug in a newer version of Pester that broke tests. The Pester team fixed this but we hadn't got round to backing the change out.

J0F3 commented:
Well I have changed the things now a little. I changed the Get-PowerPlans function to Get-PowerPlan with a -Name parameter and added a new Set-PowerPlan function. 

However in the meantime I have also made some tests with using the native Win32 API for power management (which probably also the powercfg.exe uses). I am now able to get and set power plans using the win32 API and P/Invoke. 

I think this would be actually the most robust approach and I will crate probably an second PR with this P/Invoke approach. Or should I just update this one?

J0F3 commented:
All right I just created an other PR with the approach to use the Windows APIs through some new functions. Actually just to have  both variants here to compare and then just close one and go further with the one we prefer. -> #203 

I personally prefer the variant with the Windows APIs but to be honest I have currently no idea how to write proper unit tests for the functions which call the Windows native APIs trough P/Invoke. 🙄
